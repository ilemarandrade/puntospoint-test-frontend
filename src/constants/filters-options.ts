import {
  EnumDateMainParameters,
  EnumFiltersTags,
  EnumThisMonthSubParameters,
  EnumThisWeekSubParameters,
  IFiltersByDates,
} from '../types/filters';

export const monthsSpanish = {
  [EnumThisMonthSubParameters.ALL]: 'Todo',
  [EnumThisMonthSubParameters.JANUARY]: 'Enero',
  [EnumThisMonthSubParameters.FEBRUARY]: 'Febrero',
  [EnumThisMonthSubParameters.MARCH]: 'Marzo',
  [EnumThisMonthSubParameters.APRIL]: 'Abril',
  [EnumThisMonthSubParameters.MAY]: 'Mayo',
  [EnumThisMonthSubParameters.JUNE]: 'Junio',
  [EnumThisMonthSubParameters.JULY]: 'Julio',
  [EnumThisMonthSubParameters.AUGUST]: 'Agosto',
  [EnumThisMonthSubParameters.SEPTEMBER]: 'Septiembre',
  [EnumThisMonthSubParameters.OCTOBER]: 'Octubre',
  [EnumThisMonthSubParameters.NOVEMBER]: 'Noviembre',
  [EnumThisMonthSubParameters.DECEMBER]: 'Diciembre',
};

export const weekDaysSpanish = {
  [EnumThisWeekSubParameters.ALL]: 'Todo',
  [EnumThisWeekSubParameters.MONDAY]: 'Lunes',
  [EnumThisWeekSubParameters.TUESDAY]: 'Martes',
  [EnumThisWeekSubParameters.WEDNESDAY]: 'Miércoles',
  [EnumThisWeekSubParameters.THURSDAY]: 'Jueves',
  [EnumThisWeekSubParameters.FRIDAY]: 'Viernes',
  [EnumThisWeekSubParameters.SATURDAY]: 'Sábado',
  [EnumThisWeekSubParameters.SUNDAY]: 'Domingo',
};

const createMonthFilters = () => {
  return Object.values(EnumThisMonthSubParameters).map((month) => ({
    name: month,
    label: monthsSpanish[month],
  }));
};

const createYearFilters = () => {
  const years = ['2020', '2021', '2022', '2023', '2024', '2025'];
  return [
    { name: EnumThisMonthSubParameters.ALL, label: 'Todo' },
    ...years.map((year) => ({ name: year, label: year })),
  ];
};

export const monthsFilters = createMonthFilters();

export const yearsFilters = createYearFilters();

export const monthsInNumber = Object.fromEntries(
  Object.values(EnumThisMonthSubParameters).map((month, index) => [
    month,
    index,
  ])
);

export const filtersByDates: IFiltersByDates[] = [
  {
    parameter: EnumDateMainParameters.TODAY,
    label: 'Hoy',
  },
  {
    parameter: EnumDateMainParameters.THIS_WEEK,
    label: '7D',
    subParameters: [
      ...Object.values(EnumThisWeekSubParameters).map((day) => ({
        name: day,
        label: weekDaysSpanish[day],
      })),
    ],
  },
  {
    parameter: EnumDateMainParameters.THIS_MONTH,
    label: 'Este mes',
  },
  {
    parameter: EnumDateMainParameters.THIS_SEMESTER,
    label: '6M',
    subParameters: monthsFilters,
  },
  {
    parameter: EnumDateMainParameters.YTD_YTG,
    label: 'YTD / YTG',
  },
  {
    parameter: EnumDateMainParameters.THIS_YEAR,
    label: '1A',
    subParameters: monthsFilters,
  },
  {
    parameter: EnumDateMainParameters.MAX,
    label: 'MAX',
    subParameters: yearsFilters,
  },
  {
    parameter: EnumDateMainParameters.CUSTOM,
    label: 'Personalizado',
  },
];

export const titleTableDate = {
  [EnumDateMainParameters.TODAY]: { title: 'Hoy', header: 'Horas' },
  [EnumDateMainParameters.THIS_WEEK]: { title: '7 días', header: 'Semana' },
  [EnumDateMainParameters.THIS_MONTH]: { title: 'Este mes', header: 'Días' },
  [EnumDateMainParameters.THIS_SEMESTER]: { title: '6 meses', header: 'Meses' },
  [EnumDateMainParameters.YTD_YTG]: { title: 'YTD / YTG', header: 'Meses' },
  [EnumDateMainParameters.THIS_YEAR]: { title: '1 Año', header: 'Meses' },
  [EnumDateMainParameters.MAX]: { title: 'MAX', header: 'Años' },
  [EnumDateMainParameters.CUSTOM]: {
    title: 'Personalizado',
    header: 'Personalizado',
  },
};

export const filterTags = {
  firstGroup: [
    {
      label: 'Clientes',
      value: EnumFiltersTags.CLIENTS,
      dataTestid: 'clients',
    },
    {
      label: 'Transacciones',
      value: EnumFiltersTags.TRANSACTIONS,
      dataTestid: 'transactions',
    },
  ],
  secondGroup: [
    {
      label: 'Dinero',
      value: EnumFiltersTags.MONEY,
      dataTestid: 'money',
    },
    {
      label: 'Cashback',
      value: EnumFiltersTags.CASHBACK,
      dataTestid: 'cashback',
    },
  ],
};
