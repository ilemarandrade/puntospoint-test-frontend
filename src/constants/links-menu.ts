export const linksMenu = [
  { title: 'Dashboard', url: '/' },
  { title: 'Clientes', url: '/clients' },
  { title: 'Reglas de acumulaci√≥n', url: '/rules' },
];
