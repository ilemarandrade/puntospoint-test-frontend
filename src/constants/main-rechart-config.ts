import { EnumFiltersTags } from '@/types/filters';
import { ISalesData } from '@/types/recharts';
import { LegendType } from 'recharts';

interface IMainRechartConfig {
  value: string;
  typeLegend: LegendType;
  type: 'line' | 'bar';
  color: string;
  id: EnumFiltersTags;
  label: string;
  tooltip: string;
  keyAccessor: keyof ISalesData;
}

export const mainRechartConfig: IMainRechartConfig[] = [
  {
    keyAccessor: 'totalCustomers',
    value: 'Clientes totales',
    typeLegend: 'square',
    type: 'bar',
    color: '#358DEB',
    id: EnumFiltersTags.CLIENTS,
    label: 'Total de Clientes',
    tooltip: 'Número total de clientes registrados',
  },
  {
    keyAccessor: 'newCustomers',
    value: 'Clientes nuevos',
    typeLegend: 'square',
    type: 'bar',
    color: '#EB7635',
    id: EnumFiltersTags.CLIENTS,
    label: 'Clientes Nuevos',
    tooltip: 'Número de nuevos clientes en el periodo',
  },
  {
    keyAccessor: 'purchased',
    value: 'Compraron',
    typeLegend: 'square',
    type: 'bar',
    color: '#2DCF5A',
    id: EnumFiltersTags.CLIENTS,
    label: 'Clientes que Compraron',
    tooltip: 'Clientes que realizaron al menos una compra',
  },
  {
    keyAccessor: 'notPurchased',
    value: 'No compraron',
    typeLegend: 'square',
    type: 'bar',
    color: '#EB3535',
    id: EnumFiltersTags.CLIENTS,
    label: 'Clientes que No Compraron',
    tooltip: 'Clientes que no realizaron compras',
  },
  {
    keyAccessor: 'transactions',
    value: 'Transacciones',
    typeLegend: 'square',
    type: 'bar',
    color: '#358DEB',
    id: EnumFiltersTags.TRANSACTIONS,
    label: 'Total de Transacciones',
    tooltip: 'Número total de transacciones realizadas',
  },
  {
    keyAccessor: 'totalMoney',
    value: 'Dinero total',
    typeLegend: 'line',
    type: 'line',
    color: '#EB3535',
    id: EnumFiltersTags.MONEY,
    label: 'Dinero Total',
    tooltip: 'Total de dinero generado en ventas',
  },
  {
    keyAccessor: 'sales',
    value: 'Ventas',
    typeLegend: 'line',
    type: 'line',
    color: '#7A35EB',
    id: EnumFiltersTags.MONEY,
    label: 'Ventas Totales',
    tooltip: 'Total de ventas realizadas',
  },
  {
    keyAccessor: 'returns',
    value: 'Devoluciones',
    typeLegend: 'line',
    type: 'line',
    color: '#48454E',
    id: EnumFiltersTags.MONEY,
    label: 'Devoluciones Totales',
    tooltip: 'Total de devoluciones realizadas',
  },
  {
    keyAccessor: 'cashbackGenerated',
    value: 'Cashback generado',
    typeLegend: 'line',
    type: 'line',
    color: '#EB3535',
    id: EnumFiltersTags.CASHBACK,
    label: 'Cashback Generado',
    tooltip: 'Total de cashback generado en ventas',
  },
  {
    keyAccessor: 'cashbackAccumulated',
    value: 'Cashback acumulado',
    typeLegend: 'line',
    type: 'line',
    color: '#7A35EB',
    id: EnumFiltersTags.CASHBACK,
    label: 'Cashback Acumulado',
    tooltip: 'Total de cashback acumulado por clientes',
  },
  {
    value: 'Cashback total',
    typeLegend: 'line',
    type: 'line',
    color: '#EB35AD',
    id: EnumFiltersTags.CASHBACK,
    label: 'Cashback Total',
    tooltip: 'Total de cashback entregado a los clientes',
    keyAccessor: 'totalCashback',
  },
];

export const payloadLegend = mainRechartConfig.map((item) => ({
  value: item.value,
  type: item.typeLegend,
  color: item.color,
  id: item.id,
}));
