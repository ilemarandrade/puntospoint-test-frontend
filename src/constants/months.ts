export const monthsData = [
  {
    numberMonth: 1,
    name: 'Enero',
  },
  {
    numberMonth: 2,
    name: 'Febrero',
  },
  {
    numberMonth: 3,

    name: 'Marzo',
  },
  {
    numberMonth: 4,
    name: 'Abril',
  },
  {
    numberMonth: 5,
    name: 'Mayo',
  },
  {
    numberMonth: 6,
    name: 'Junio',
  },
  {
    numberMonth: 7,
    name: 'Julio',
  },
  {
    numberMonth: 8,
    name: 'Agosto',
  },
  {
    numberMonth: 9,
    name: 'Septiembre',
  },
  {
    numberMonth: 10,
    name: 'Octubre',
  },
  {
    numberMonth: 11,
    name: 'Noviembre',
  },
  {
    numberMonth: 12,
    name: 'Diciembre',
  },
];
