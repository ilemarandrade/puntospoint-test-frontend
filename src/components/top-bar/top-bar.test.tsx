import React from 'react';
import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import TopBar from '.';
import { linksMenu } from '@/constants/links-menu';

jest.mock('@mui/material', () => {
  return {
    ...jest.requireActual('@mui/material'),
    useMediaQuery: () => false,
  };
});

describe('TopBar Component', () => {
  test('renders links with correct href attributes', () => {
    const { container } = render(<TopBar />);

    linksMenu.forEach((link) => {
      const anchorElement = screen.getByText(link.title);
      expect(anchorElement).toBeInTheDocument();
      expect(anchorElement).toHaveAttribute('href', link.url);
    });

    expect(container).toMatchSnapshot();
  });

  test('opens and closes the menu', async () => {
    render(<TopBar />);

    expect(screen.queryByText('Editar perfil')).not.toBeInTheDocument();
    expect(screen.queryByText('Cerrar Sesion')).not.toBeInTheDocument();

    const button = screen.getByTestId('ExpandMoreRoundedIcon');

    fireEvent.click(button);

    expect(screen.queryByText('Editar perfil')).toBeVisible();
    expect(screen.queryByText('Cerrar Sesion')).toBeVisible();

    fireEvent.click(screen.getByTestId('LogoutRoundedIcon'));

    await waitFor(() => {
      expect(screen.queryByText('Editar perfil')).not.toBeVisible();
    });
    expect(screen.queryByText('Cerrar Sesion')).not.toBeVisible();
  });
});
