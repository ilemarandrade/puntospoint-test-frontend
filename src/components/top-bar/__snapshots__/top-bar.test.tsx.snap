// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar Component renders links with correct href attributes 1`] = `
<div>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionSticky css-1d63wei-MuiPaper-root-MuiAppBar-root"
    style="--Paper-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);"
  >
    <div
      class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row MuiGrid2-spacing-xs-2 css-ubb9a5-MuiGrid2-root"
    >
      <div
        class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row MuiGrid2-grid-xs-0 MuiGrid2-grid-sm-0 MuiGrid2-grid-lg-3 css-1a0qr9e-MuiGrid2-root"
      >
        <img
          alt="Logo"
          class="w-6/12"
          data-nimg="1"
          decoding="async"
          height="100"
          loading="lazy"
          src="/puntospoint-logo.svg"
          style="color: transparent;"
          width="100"
        />
      </div>
      <div
        class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row MuiGrid2-grid-xs-grow MuiGrid2-grid-lg-6 css-1dw3rk4-MuiGrid2-root"
      >
        <a
          class="text-primary rounded-[100px] h-[40px] gap-2 py-2.5 px-6 font-medium text-sm leading-5 tracking-tight text-center"
          href="/dashboard"
        >
          Dashboard
        </a>
        <a
          class="text-primary rounded-[100px] h-[40px] gap-2 py-2.5 px-6 font-medium text-sm leading-5 tracking-tight text-center"
          href="/clients"
        >
          Clientes
        </a>
        <a
          class="text-primary rounded-[100px] h-[40px] gap-2 py-2.5 px-6 font-medium text-sm leading-5 tracking-tight text-center"
          href="/rules"
        >
          Reglas de acumulaci√≥n
        </a>
      </div>
      <div
        class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-auto MuiGrid2-grid-lg-3 css-174tjep-MuiGrid2-root"
      >
        <div
          class="w-full h-full flex items-center justify-center gap-1"
        >
          <p
            class="font-medium text-base leading-6 text-center align-middle"
          >
            Pamela Rojas Gonzalez
          </p>
          <button
            aria-haspopup="true"
            class="MuiButtonBase-root css-qn10b5-MuiButtonBase-root"
            id="basic-button"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
              data-testid="ExpandMoreRoundedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.88 9.29 12 13.17 8.12 9.29a.996.996 0 0 0-1.41 0c-.39.39-.39 1.02 0 1.41l4.59 4.59c.39.39 1.02.39 1.41 0l4.59-4.59c.39-.39.39-1.02 0-1.41-.39-.38-1.03-.39-1.42 0"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </header>
</div>
`;
